core/src/main/java/fr/sncf/osrd/infra/api/tracks/undirected/SpeedLimits.java

```
@@ -30,7 +41,12 @@ public static SpeedLimits from(RJSSpeedSection rjsSpeedSection) {
             validateSpeedLimit(limit);
         }
 
-        return new SpeedLimits(speedLimit, speedLimitByTag);
+        var speedLimitOnRoute = new HashMap<String, Double>();
+        if (rjsSpeedSection.onRoutes != null) {
+            speedLimit = Double.POSITIVE_INFINITY;
```

When a route is set, we "empty" the speed limit and only fill values in the limits per route field?

That makes sense, but then it's inconsistent if both routes and speed limits by tag are set. I assume it shouldn't happen, but maybe we should throw an error if it does anyway

---

Done